src/math_funcs.jl

```diff
@@ -0,0 +1,77 @@
+# define maths function for rotations types
+import Base.(*), Base.(.*), Base.(-)
+
+function add_maths(rot_type)
+
+    # create an expression for a tuple X[1], X[2], etc...
+    fields = fieldnames(rot_type)
+
+    # the code block we are creating
+    qb = quote; end
+
+    # allow rotating a Vec{3, ...} using *    
+    if !(rot_type <: FixedArray)
+        qn = quote
+            (*){T <: Real}(R::$(rot_type), X::Vec{3,T}) = rotate_point(R, X)
```

Comment:
After we talked about it I realized its templates for Vectors with 3 real elements.  Does multiplication have only one definition for Reals?